new-log-viewer/src/components/DropFileContainer/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+    useContext,
+    useState,
+} from "react";
+
+import {StateContext} from "../../contexts/StateContextProvider";
+import {CURSOR_CODE} from "../../typings/worker";
+
+import "./index.css";
+
+
+interface DropFileContextProviderProps {
+    children: React.ReactNode;
+}
+
+/**
+ * A container element to add drag & drop functionality to the child elements.
+ *
+ * @param props
+ * @param props.children
+ * @return
+ */
+const DropFileContainer = ({children}: DropFileContextProviderProps) => {
+    const {loadFile} = useContext(StateContext);
+    const [isFileHovering, setIsFileHovering] = useState(false);
+
+    /**
+     * Handler for a drag event.
+     *
+     * @param ev
+     */
+    const handleDrag = (ev: React.DragEvent<HTMLDivElement>) => {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        if ("dragenter" === ev.type) {
+            setIsFileHovering(true);
+        } else if ("dragleave" === ev.type) {
+            const {bottom, left, right, top} = ev.currentTarget.getBoundingClientRect();
+            if (ev.clientX >= left && ev.clientX <= right &&
+                ev.clientY >= top && ev.clientY <= bottom) {
+                return;
+            }
+
+            setIsFileHovering(false);
+        }
+    };
+
+    /**
+     * Handler for a drop event. handleFileDrop callback is used
+     * to load the dropped file into the viewer.
```

Comment:
I guess the second sentence is obsolete?

Reply:
The whole JSDoc block was removed in the latest commits.